services/src/main/java/org/keycloak/protocol/oidc/AccessTokenIntrospectionProvider.java

```diff
@@ -107,6 +108,12 @@ public Response introspect(String token, EventBuilder eventBuilder) {
 
             tokenMetadata.put("active", userSession != null);
 
+            // if consumer requests application/jwt return a JWT representation of the introspection contents in an jwt field
+            if (org.keycloak.utils.MediaType.APPLICATION_JWT.equals(session.getContext().getRequestHeaders().getHeaderString(HttpHeaders.ACCEPT))) {
```

Comment:
This allows all legacy callers of the token introspection endpoint to translate a lightweight access token into a “full” access token. Depending on how extensively the token introspection endpoint is exposed, it could be useful to check whether the client is actually allowed to do this.

A configurable client attribute such as `token_introspection_jwt_response_enabled` “JWT in Token Introspection Response allowed (on/off)” could be helpful here.

Only clients that have `token_introspection_jwt_response_enabled` active may then also generate JWT tokens in the introspection response. This could then only be active for “internal” clients, for example.